src/edu/cs505/g5/SubjectDHTSensor.java

```diff
@@ -0,0 +1,46 @@
+p
+package edu.ccsu.cs505.g5;
+/**
+ *
+ * @author CS505-Group5
+ */
+import java.util.Vector;
+/**
+ * This class is the Subject class in the observer pattern which will notify the observers if the temperature goes beyond the limit-set
+ */
+public class SubjectDHTSensor {
+    
+    /**
+     * Because the SubjectDHTSensor calls the interface Observer and all other classes that implement that interface as well
+     */
+        private ObserverLED ledObserver = new ObserverLED();

```

Comment:
The Subject base class should NOT have any references to the concrete observers.  It should only have a list of ObserverDHTSensor interfaces.  Based on this design it would force anybody that wanted to use your component to have all of these observers.  You instead want to have none, and allow them to add what they want.

Reply:
modified